replace numpy.NAN with numpy.nan

Update code to use [`numpy.nan`](https://numpy.org/doc/stable/reference/constants.html#numpy.nan) instead of `NAN`
```
FAILED tests/fileformats/test_xvg.py::TestXVG_array::test_decimate[circmean] - AttributeError: module 'numpy' has no attribute 'NAN'
FAILED tests/fileformats/test_xvg.py::test_break_array - AttributeError: module 'numpy' has no attribute 'NAN'
```

--- a/CHANGES
+++ b/CHANGES
@@ -12,6 +12,7 @@ orbeckst
   (#282)
 * fixed reporting of failures of GromacsCommand and DeprecationWarnings for use
   of \s in regular expression strings (#285)
+* fixed use of outdated numpy.NAN | INF by using nan | inf (#290)
 
 
 2024-06-15      0.9.0

--- a/gromacs/fileformats/xvg.py
+++ b/gromacs/fileformats/xvg.py
@@ -214,7 +214,7 @@ import gromacs.collections
 class XVG(utilities.FileUtils):
     """Class that represents the numerical data in a grace xvg file.
 
-    All data must be numerical. :const:`NAN` and :const:`INF` values are
+    All data must be numerical. :const:`nan` and :const:`inf` values are
     supported via python's :func:`float` builtin function.
 
     The :attr:`~XVG.array` attribute can be used to access the the
@@ -1326,12 +1326,12 @@ def break_array(a, threshold=numpy.pi, other=None):
     mask = numpy.zeros_like(b, dtype=bool)
     mask[b_breaks] = True
     b[~mask] = a
-    b[mask] = numpy.NAN
+    b[mask] = numpy.nan
 
     if other is not None:
         c = numpy.empty_like(b)
         c[~mask] = other
-        c[mask] = numpy.NAN
+        c[mask] = numpy.nan
         ma_c = numpy.ma.array(c, mask=mask)
     else:
         ma_c = None

--- a/tests/fileformats/test_xvg.py
+++ b/tests/fileformats/test_xvg.py
@@ -87,7 +87,7 @@ class TestXVG_array(object):
 
 def test_break_array():
     angles = np.pi * np.array([-1.9, -1, -1, -0.5, 0, 0.9, 1.5, 2, -2, -1.4])
-    expected = np.pi * np.array([-1.9, -1, -1, -0.5, 0, 0.9, 1.5, 2, np.NAN, -2, -1.4])
+    expected = np.pi * np.array([-1.9, -1, -1, -0.5, 0, 0.9, 1.5, 2, np.nan, -2, -1.4])
     other = np.ones_like(angles)
     ma, mother = gromacs.fileformats.xvg.break_array(
         angles, threshold=np.pi, other=other

